ts/packages/anchor/src/coder/borsh/instruction.ts

```diff
@@ -56,11 +56,15 @@ export class BorshInstructionCoder implements InstructionCoder {
   /**
    * Encodes a program instruction.
    */
-  public encode(ixName: string, ix: any): Buffer {
-    return this._encode(SIGHASH_GLOBAL_NAMESPACE, ixName, ix);
+  public encode(ixName: string, ix: any, discriminator?: Buffer): Buffer {
+    return this._encode(
+      discriminator ?? sighash(SIGHASH_GLOBAL_NAMESPACE, ixName),
+      ixName,
+      ix
+    );
   }
 
-  private _encode(nameSpace: string, ixName: string, ix: any): Buffer {
+  private _encode(discriminator: Buffer, ixName: string, ix: any): Buffer {
```

Comment:
Could you keep the parameter order the same as `encode`?

```suggestion
  private _encode(ixName: string, ix: any, discriminator: Buffer): Buffer {
```